db/dynamodb-global-table.yml

```diff
@@ -0,0 +1,337 @@
+                - { Key: prx:dev:family, Value: !Ref FamilyTag }
+                - { Key: prx:dev:application, Value: !Ref ApplicationTag }
+            - !Ref AWS::NoValue
+      StreamSpecification:
+        StreamViewType: NEW_AND_OLD_IMAGES
+      TableName: !Ref TableName
+      TimeToLiveSpecification:
+        AttributeName: !If [HasExpirationField, !Ref ExpirationField, !Ref AWS::NoValue]
+        Enabled: !If [HasExpirationField, true, false]
+      WriteProvisionedThroughputSettings: !If
+        - HasProvisionedBilling
+        - WriteCapacityAutoScalingSettings:
+            MaxCapacity: !Ref WriteCapacityMax
+            MinCapacity: !Ref WriteCapacityMin
+            SeedCapacity: !Ref WriteCapacitySeed
+            TargetTrackingScalingPolicyConfiguration:
+              ScaleInCooldown: !Ref AutoScalingCooldown
+              ScaleOutCooldown: !Ref AutoScalingCooldown
+              TargetValue: !Ref WriteUtilizationScalingTarget
+        - !Ref AWS::NoValue
+
+  # allow other accounts access to this table
+  CrossAccountAccessRole:
+    Type: AWS::IAM::Role
+    Condition: HasForeignAccount
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRole
+            Principal: !If
+              - HasForeignRole
+              - AWS: !Sub arn:aws:iam::${ForeignAccountId}:role/${ForeignRoleName}
+              - AWS: !Ref ForeignAccountId
+      Path: /
+      Policies:
+        - PolicyName: DynamoDBWritePolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:BatchGetItem
+                  - dynamodb:BatchWriteItem
+                  - dynamodb:ConditionCheck
+                  - dynamodb:DeleteItem
+                  - dynamodb:DescribeTable
+                  - dynamodb:DescribeTimeToLive
+                  - dynamodb:GetItem
+                  - dynamodb:PutItem
+                  - dynamodb:Query
+                  - dynamodb:UpdateItem
+                Resource:
+                  - !GetAtt DynamoDBTable.Arn
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:family, Value: !Ref FamilyTag }
+        - { Key: prx:dev:application, Value: !Ref ApplicationTag }
+
+  # Alarms
+  # TODO: provisioned billing might benefit from alarms - but they might have
+  # to exist in multiple regions to cover all these tables
+Outputs:
+  RoleArn:
+    Condition: HasForeignAccount
+    Description: The created DynamoDB table ARN
+    Value: !GetAtt CrossAccountAccessRole.Arn
+  TableArn:
+    Description: The created DynamoDB table ARN
```

Comment:
This would be the ARN for just _one_ of the tables that's created, right?

Reply:
Oh, yeah, I guess it is.  Maybe I'll change this output to `TableName` anyways ... that's what our lambdas take as an input.  The table name, region, and access role arn.